### PR TITLE
Initialise frontend sets from pandas dataframes

### PR DESCRIPTION
**Pull request by demianw**

### CODE REVIEW

File: examples/plot_neurosynth_implementation.py [hunk 1]
```diff
@@ -90,18 +90,17 @@ def agg_create_region_overlay(
...
+docs = nl.add_uniform_probabilistic_choice_over_set(ns_docs, name="docs")
 
 
+# %%
 ###############################################################################
 # Probabilistic program and querying
 
 with nl.scope as e:
-    e.vox_activation[e.i, e.j, e.k, e.d] = e.activations[
+    e.vox_activation[e.i, e.j, e.k, e.d] = e.activations(
```
Comment (tgy) [inline examples/plot_neurosynth_implementation.py:103]: why this change of syntax? just wondering

File: examples/plot_neurosynth_implementation.py [hunk 2]
```diff
@@ -90,18 +90,17 @@ def agg_create_region_overlay(
...
 # Loading the database
 
-activations = nl.add_tuple_set(ns_database.values, name="activations")
-terms = nl.add_tuple_set(ns_terms.values, name="terms")
-docs = nl.add_uniform_probabilistic_choice_over_set(
-    ns_docs.values, name="docs"
-)
+activations = nl.add_tuple_set(ns_database, name="activations")
+terms = nl.add_tuple_set(ns_terms, name="terms")
+docs = nl.add_uniform_probabilistic_choice_over_set(ns_docs, name="docs")
```
Comment (tgy) [inline examples/plot_neurosynth_implementation.py:95]: this looks much cleaner than having to suffix a `.values`!

**Review by tgy — APPROVED** [review]
Merged this with my example's branch and can confirm this works! Thanks

File: examples/plot_neurosynth_implementation.py (hunk 1 above)
Comment (demianw) [reply examples/plot_neurosynth_implementation.py:103]: No, both work. I think it's nicer if it's `()` on the body of the rules but open to discussion

File: examples/plot_neurosynth_implementation.py (hunk 2 above)
Comment (demianw) [reply examples/plot_neurosynth_implementation.py:95]: Yup! Thanks for pushing for it!

File: examples/plot_neurosynth_implementation.py (hunk 1 above)
Comment (tgy) [reply examples/plot_neurosynth_implementation.py:103]: alright, that works for me. will adapt other examples to use this syntax
